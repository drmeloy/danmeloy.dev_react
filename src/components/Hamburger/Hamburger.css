.BurgerIcon {
  position: fixed;
  right: 1%;
  z-index: 2;
  display: none;

  @media (max-width: 1025px){
    display: inline;
  }
}

.BurgerNav {
  position: fixed;
  right: 0;
  padding-top: 8%;
  width: 25%;
  height: 100vh;
  background-color: #00000099;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  z-index: 1;
  animation: slideIn .5s;
  display: none;

  .Link {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  img {
    height: 5vh;
  }

  img.active {
    display: none;
  }

  p {
    color: white;
    font-family: 'Abril Fatface', cursive;
    font-size: 0.6em;
  }

  .Active {
    img.inactive {
      display: none;
    }

    img.active {
      display: block;
    }

    p {
      color: #fff700;
    }
  }

  @media (min-width: 721px) and (max-width: 1025px){
    p {
      font-size: .8em;
    }

    width: 16%;
  }

  @media (max-width: 1025px){
    display: flex;
  }

  @keyframes slideIn {
    from { transform: translateX(100%); }
    to { transform: translateX(0); }
  }
}

.SlideOut {
  animation: slideOut .5s;

  @keyframes slideOut {
    from { transform: translateX(0); }
    to { transform: translateX(100%); }
  }
}