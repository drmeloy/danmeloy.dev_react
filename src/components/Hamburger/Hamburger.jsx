import React, { useState, useRef } from 'react';
import styles from './Hamburger.css';
import Scroll from 'react-scroll';
const Link = Scroll.Link;
import Burger from 'react-css-burger';

export default function Hamburger(){
  const [open, setOpen] = useState(false);
  const sidebar = useRef();

  const clickMenu = () => {
    console.log(sidebar.classList);
    
    if(!open){
      setOpen(!open);
    } else {
      sidebar.current.classList.add(styles.SlideOut);
      setTimeout(() => {
        setOpen(!open);
      }, 500);
    }
    
  };

  return (
    <>
      <div className={styles.BurgerIcon}>
        <Burger 
          onClick={clickMenu}
          active={open}
          burger='emphatic'
          color={'#C0C0C0'}
          scale={.5}
          marginTop={0}
          marginLeft={0}
        />
      </div>
      {open ? <nav ref={sidebar} className={styles.BurgerNav}>
        <Link
          to='Home'
          spy={true}
          smooth={true}
          duration={800}
          className={styles.Link}
          activeClass={styles.Active}>
          <img className={styles.inactive} src='../../../public/assets/home.png'></img>
          <img className={styles.active} src='../../../public/assets/home_active.png'></img>
          <p>Home</p>
        </Link>
        <Link
          to='About'
          spy={true}
          hashSpy={true}
          smooth={true}
          duration={800}
          className={styles.Link}
          activeClass={styles.Active}>
          <img className={styles.inactive} src='../../../public/assets/about.png'></img>
          <img className={styles.active} src='../../../public/assets/about_active.png'></img>
          <p>About</p>
        </Link>
        <Link
          to='Work'
          spy={true}
          hashSpy={true}
          smooth={true}
          duration={800}
          className={styles.Link}
          activeClass={styles.Active}>
          <img className={styles.inactive} src='../../../public/assets/work.png'></img>
          <img className={styles.active} src='../../../public/assets/work_active.png'></img>
          <p>Work</p>
        </Link>
        <Link
          to='Background'
          spy={true}
          hashSpy={true}
          smooth={true}
          duration={800}
          className={styles.Link}
          activeClass={styles.Active}>
          <img className={styles.inactive} src='../../../public/assets/background.png'></img>
          <img className={styles.active} src='../../../public/assets/background_active.png'></img>
          <p>Background</p>
        </Link>
        <Link
          to='Contact'
          spy={true}
          hashSpy={true}
          smooth={true}
          duration={800}
          className={styles.Link}
          activeClass={styles.Active}>
          <img className={styles.inactive} src='../../../public/assets/contact.png'></img>
          <img className={styles.active} src='../../../public/assets/contact_active.png'></img>
          <p>Contact</p>
        </Link>
      </nav>
        : ''}
    </>
  );
}
