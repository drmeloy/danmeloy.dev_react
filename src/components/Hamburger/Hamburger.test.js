import React from 'react';
import { shallow } from 'enzyme';
import Hamburger from './Hamburger';

describe('Hamburger component', () => {
  it('matches snapshop', () => {
    const wrapper = shallow(<Hamburger />);
    expect(wrapper).toMatchSnapshot();
  });
});
