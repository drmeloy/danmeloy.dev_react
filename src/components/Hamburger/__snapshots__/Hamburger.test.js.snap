// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger component matches snapshop 1`] = `
<Fragment>
  <div
    className="BurgerIcon"
  >
    <Burger
      active={false}
      burger="emphatic"
      color="#C0C0C0"
      hoverOpacity={1}
      marginLeft="0"
      marginTop="0"
      onClick={[Function]}
      scale={0.5}
    />
  </div>
</Fragment>
`;
